ethereum-client/src/events.rs

```diff
@@ -312,6 +312,8 @@ pub fn decode_event(topics: Vec<String>, data: String) -> Result<EthereumEvent,
         .iter()
         .map(|topic| decode_topic(topic).ok_or(EventError::InvalidTopic))
         .collect::<Result<Vec<ethabi::Hash>, _>>()?;
+
+    // xxx todo:wn note - this is a strange match - need to check the syntax on this and see what it really means.....
```

Comment:
You mean using the guards?

Reply:
Ok i just did the todo. It turns out that the thing to the left of the `=>` must be a `PATTERN` and here are the docs for this kind of pattern

https://doc.rust-lang.org/book/ch18-03-pattern-syntax.html#extra-conditionals-with-match-guards

I _thought_ `t if *t == *LOCK_EVENT_TOPIC` was being evaluated as an expression somehow with that expressions output resulting in the LHS of the match. Importantly `t if *t == *LOCK_EVENT_TOPIC` is NOT in fact an expression and can not be.